submissions/natashafir/friends-app/index.js

```diff
@@ -0,0 +1,107 @@
+const URL = 'https://randomuser.me/api/?results=50';
+const CARDS = document.querySelector('.cards');
+let filterBy = 'all';
+let allFriends = [];
+let friends = [];
+let sortedFriends;
+const input = document.querySelector('input');
+
+function initialApp() {
+    fetch(URL)
+        .then(response => response.json())
+        .then((data) => (allFriends = data.results))
+        .then(() => userTemplate(allFriends))
+        .then(()=> renderUsers(allFriends))
+}
+
+function userTemplate(array) {
+    allFriends = array.map(user => {
+            return {
+                name: `${user.name.first} ${user.name.last}`,
+                age: user.dob.age,
+                gender: user.gender,
+                photo: user.picture.large
+            }
+        });
+}
+
+function createCardItem(item) {
+    const BR = document.createElement("br")
+    const cardWrapper = document.createElement('div');
+    const img = document.createElement('img');
+    const name = document.createElement('p');
+    const userName = document.createTextNode(item.name);
+    const age = document.createElement('p');
+    const userAge = document.createTextNode(item.age);
+    const gender = document.createElement('p');
+    const userGender = document.createTextNode(item.gender);
+
+    cardWrapper.classList.add('card-wrapper');
+    img.classList.add('img-card');
+
+    CARDS.appendChild(cardWrapper);
+    cardWrapper.appendChild(img);
+    cardWrapper.appendChild(name);
+    name.appendChild(userName);
+    name.appendChild(BR);
+    cardWrapper.appendChild(age);
+    name.appendChild(userAge);
+    cardWrapper.appendChild(gender);
+
+    img.src = item.photo;
+}
+
+
+document.querySelector(".gender").addEventListener("click", filterByGender);
+document.querySelector(".age").addEventListener("click", filterByGender);
+document.querySelector(".name").addEventListener("click", filterByGender);
```

Comment:
`filterByGender` used for sorting by age, which is wrong.

Single responsibility - your functions should do only one job. As an example function in which you fetch users should only fetch them and not render, transform or process them in other ways. All these things should be done in another place, outside your function. 

The same applied to the sort function, which usually does all types of sorting 😉

Reply:
Also, consider using of event delegation